spec/models/invoices/pdf_spec.rb

```diff
@@ -3,36 +3,51 @@
 describe Invoices::PDF do
   let(:invoice) { create :invoice, :default_associations }
   let(:pdf) { described_class.new(invoice) }
+  let(:persisted_pdf_path) { "#{Global.invoices.persisted_pdfs_directory}/#{invoice.id}.pdf" }
 
-  describe 'create' do
-    let(:main_pdf_generator) { double }
-    let(:pdf_double) { OpenStruct.new id: 1337, path: 'main.pdf' }
+  describe '#read' do
+    context 'when pdf is persisted' do
+      before do
+        pdf.persist!
+      end
 
-    before do
-      expect(Invoices::PDFGenerators::Main).to receive(:new).with(invoice).and_return main_pdf_generator
-      expect(main_pdf_generator).to receive(:pdf_path).and_return 'main.pdf'
-      expect(main_pdf_generator).to receive(:create)
-      expect(File).to receive(:new).with('main.pdf').and_return pdf_double
+      it 'returns persisted pdf' do
+        expect(pdf.read).to eq File.open(persisted_pdf_path).read
+      end
     end
 
-    describe '#create_temporary' do
-      subject { pdf.create_temporary }
+    context 'when pdf is not persisted' do
+      before do
+        pdf.remove_persisted_pdf
+      end
 
-      it 'returns created pdf path' do
-        expect(subject).to eq('main.pdf')
+      it 'returns newly rendered pdf' do
+        expect(pdf.read).to include 'PDF'
       end
     end
+  end
 
-    describe '#create_persisted' do
-      subject { pdf.create_persisted }
+  describe '#persist!' do
+    before do
+      pdf.remove_persisted_pdf
+    end
 
-      it 'copies temporary created pdf to persisted location' do
-        expect(FileUtils).to receive(:cp).with(
-          'main.pdf', "#{Global.invoices.persisted_pdfs_directory}/#{invoice.id}.pdf"
-        )
+    it 'perists pdf' do
+      expect do
+        pdf.persist!
+      end.to change { File.exists?(persisted_pdf_path) }.from(false).to(true)
```

Comment:
Lint/DeprecatedClassMethods: `File.exists?` is deprecated in favor of `File.exist?`.